### PR TITLE
Add `subxt` feature in  `subxt-signer` crate to default features

### LINKED ISSUE #1189: Docs: clarify "subxt" feature in subxt-signer
It's come up a couple of times now where people have used subxt-signer with subxt and have run into an error about the signer trait not being implemented.

This is resolved by adding the "subxt" feature on the subxt-signer dep to tell it to play nicely with subxt.

We should either:
- make the feature be enabled by default because 99% of the time it'll be used with subxt anyway (I think this is my preference), or
- Clarify in the docs that the subxt feature should be enabled.

### PR DESCRIPTION
**Pull request by tadeohepperle**
Fixes #1189 .

Small PR to simplify use of subxt-signer to avoid confusion.

### CODE REVIEW

**Review by jsdw — APPROVED** [review]
Looks good to me! A nice little QoL tweak

**Review by niklasad1 — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
Nice! 👍

**Comment by jsdw** [comment]
I wonder whether, to fix the feature flag compile issue, we should add no-default-features to the subxt dep in subxt-signer (it shouldn't care, after all, which subxt features are enabled). That might lead to other issues htough; needs some playing with!

File: signer/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ description = "Sign extrinsics to be submitted by Subxt"
 keywords = ["parity", "subxt", "extrinsic", "signer"]
 
 [features]
-default = ["sr25519", "ecdsa"]
+default = ["sr25519", "ecdsa", "subxt", "native"]
```
Comment (jsdw) [inline signer/Cargo.toml:18]: I don't really understand this change; it means that signer by default will be "native" and then to be web you need to disable default features and then select eg subxt,web?

I think it'd be better to do as subxt and ask the depends library to pick which one it wants explicitly. This way, libraries can (and should) defer the choice unless it's relevant to them :)

Comment (tadeohepperle) [reply signer/Cargo.toml:18]: @jsdw My reasoning was that in 90% of cases people will just want to use it with native anyway. But you are right, I will change it to be similar to subxt in that an explicit choice has to be made.

Comment (jsdw) [reply signer/Cargo.toml:18]: Yeah, I get that; my main issue is that feature flags are additive and so the choice to enable web or native should be made at the last possible moment; enabling it by default makes it more likely that something depending on subxt somewhere in the tree and not thinking about features will screw up somebody trying to compile instead to web or whatever :)

**Pull request merged**